src/main/java/ru/yandex/practicum/filmorate/service/DirectorService.java

```diff
@@ -0,0 +1,66 @@
+package ru.yandex.practicum.filmorate.service;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+import ru.yandex.practicum.filmorate.exceptions.NotFoundEntityByIdException;
+import ru.yandex.practicum.filmorate.model.Director;
+import ru.yandex.practicum.filmorate.storage.director.DirectorStorage;
+
+import java.util.List;
+
+@Slf4j
+@Service
+public class DirectorService {
+    private final DirectorStorage directorDbStorage;
+
+    public DirectorService(@Qualifier("DirectorDbStorage") DirectorStorage directorDbStorage) {
+        this.directorDbStorage = directorDbStorage;
+    }
+
+    public List<Director> getDirectors() {
+        return directorDbStorage.getDirectors();
+    }
+
+    public Director getDirectorById(long id) {
+        return directorDbStorage.getDirectorById(id).orElseThrow(() -> {
+                    log.warn("DirectorService, getDirectorById, режисер с id {} не найден", id);
+                    throw new NotFoundEntityByIdException("Режиссер не найден", id);
```

Comment:
В orElseThrow не нужно использовать throw, тут ожидается возврат объекта исключения.
Правильно будет вернуть новое исключение:

```suggestion
                    return new NotFoundEntityByIdException("Режиссер не найден", id);
```